src/main/java/application/rest/v1/configmaps/SectionConfigMapProcessor.java

```diff
@@ -63,58 +64,106 @@
+        JsonObject map = null;
+
         if (Logger.isDebugEnabled()) {
-            Logger.log(className, "getConfigMap", Logger.LogType.DEBUG, "For component=" + name);
+            Logger.log(className, "getConfigMap", Logger.LogType.DEBUG,
+                   "\n component namespace = " + namespace +
+                   "\n component apiVersion = " + apiVersion +
+                   "\n component name = " + name +
+                   "\n component subkind= " + subkind +
+                   "\n component kind = " + kind + 
+                   "\n component ownerReferences = " + ConfigMapProcessor.ownerRefArrayToString(owners));
         }
 
-        if (name != null && !name.isEmpty()) {       
-            if (builder.getConfigMap().entrySet().size() == 0) {
-                JsonObject map = getConfigMap(client, sectionNameWithKind);
-                
-                if (map != null) {
-                    builder.merge(map);      
-                } else {
-                    if (Logger.isDebugEnabled()) {
-                        Logger.log(className, "getConfigMap", Logger.LogType.DEBUG, "Map is null.");
-                    }
-                }
-            }
+        if ((apiVersion == null) || (apiVersion.isEmpty())) {
+            if (Logger.isDebugEnabled()) 
+                Logger.log(className, "getConfigMap", Logger.LogType.DEBUG, "component apiVersion is null or empty.");
+        } else if (name != null && !name.isEmpty()) { 
+            // Get details section configmaps available in a cluster
+            KindActionMappingProcessor kam =
+                new KindActionMappingProcessor(namespace, owners, apiVersion, name, subkind, kind);
+            String configMapName = this.sectionNameWithKind + ConfigMapProcessor.getConfigMapNameSuffix(name, subkind);
+            map = getConfigMap(client, kam, namespace, configMapName, builder);
         }
         
-        return builder.getConfigMap();
+        JsonObject configMapFound = builder.getConfigMap();
+        if (Logger.isExitEnabled()) 
+            Logger.log(className, "getConfigMap", Logger.LogType.EXIT, "configMap found = " + configMapFound);
+        return configMapFound;
     } 
-         
 
-    private JsonObject getConfigMap(ApiClient client, String configMapName) {
+    private JsonObject getConfigMap(ApiClient client, KindActionMappingProcessor kam, String namespace,
+                                    String configMapName, ConfigMapBuilder builder) {
+        if (Logger.isEntryEnabled()) {
+            Logger.log(className, "getConfigMap", Logger.LogType.ENTRY, "For namespace=" + namespace + 
+                   ", configMapName=" + configMapName);
+        }
+
         // Return the map from the local cache if it's been previously loaded.
-        if (kappnavNSMapCache.containsKey(configMapName)) {
-            return kappnavNSMapCache.get(configMapName);
+        final boolean isGlobalNS = GLOBAL_NAMESPACE.equals(namespace);
```

Comment:
This ties back to first comment at top of this page:  need to revisit configmap caching. Why kappnav namespace only?  I think we need to open it up to any/all namespaces that contribute a config map as per KAM mappings.

Reply:
kappnav namespace only so it does not cache for the actdev tool, which runs in the actdev namespace.